### PR TITLE
net: Minor accumulated cleanups

### PR DESCRIPTION
**Pull request by tjps**
From now-derelict larger changes I had been working on, here are a series of DRY refactors/cleanups.  Net loss of 35 lines of code - a small step in the good fight.

In particular I think operator!= should only ever be implemented as a negation of operator==.  Lower chance for errors, and removes the possibility of divergent behavior.

### CODE REVIEW

**Comment by fanquake** [comment]
Looks like this needs a rebase.

**Review by laanwj — APPROVED** [review]
LGTM one small suggestion

**Comment by practicalswift** [comment]
Nice cleanups!

Concept ACK

**Comment by theuni** [comment]
Looks good to me too other than what @laanwj pointed out.

File: src/compat.h [hunk 1]
```diff
@@ -96,6 +96,12 @@ typedef int32_t ssize_t;
 size_t strnlen( const char *start, size_t max_len);
 #endif // HAVE_DECL_STRNLEN
 
+#ifndef WIN32
+typedef void* sockopt_arg_t;
```
Comment (sipa) [inline src/compat.h]: Type names ending in `_t` are technically reserved by POSIX, so risk colliding with system types on some architectures.

**Review by sipa — COMMENTED** [review]
utACK 27213212dde0ae5cb7abc10c347c663b34aa0469, one nit.

**Comment by laanwj** [comment]
Looks like this is ready for merge apart from @sipa's nit...

**Pull request merged**